tests/lib/file-finder.js

```diff
@@ -88,6 +88,22 @@ describe("FileFinder", function() {
                 assert.equal(actual, expected);
             });
         });
+
+        describe("Not consider directory with expected file names", function() {
+
+            it("should not be found, and an empty array returned", function() {
+                var expected = path.join(process.cwd(), "package.json");
+                process.chdir(fileFinderDir);
+                var finder = new FileFinder("package.json");
```

Comment:
I'm still very uncomfortable with this. There has to be a way to stimulate changing cwd without actually needing to change it. Maybe pass in to File Finder some how?